src/rulesets/Base/array.jl

```diff
@@ -1,3 +1,42 @@
+#####
+##### constructors
+#####
+
+ChainRules.@non_differentiable (::Type{T} where {T<:Array})(::UndefInitializer, args...)
+
+function rrule(::Type{T}, x::AbstractArray) where {T<:Array}
+    project_x = ProjectTo(x)
+    Array_pullback(ȳ) = (NoTangent(), project_x(ȳ))
+    return T(x), Array_pullback
+end
+
+#####
+##### `vect`
+#####
+
+@non_differentiable Base.vect()
+
+# Don't worry about projection here. The data passes straight through, so if a cotangent has
+# the wrong type for some reason, it must be the fault of another rule somewhere.
+function rrule(::typeof(Base.vect), X::Vararg{T, N}) where {T, N}
+    function vect_pullback(ȳ)
+        X̄ = ntuple(n -> ȳ[n], N)
+        return (NoTangent(), NTuple{N}(ȳ)...)
+    end
```

Comment:
This makes the same tuple twice, could just be:
```suggestion
    vect_pullback(ȳ) = (NoTangent(), NTuple{N}(ȳ)...)
```